oemof_examples/oemof.solph/v0.4.x/dual_variables/simple_dispatch_with_duals.py

```diff
@@ -0,0 +1,188 @@
+energysystem.add(Sink(label="excess_el", inputs={bel: Flow()}))
+# shortage_el = Source(label='shortage_el',
+#                      outputs={bel: Flow(variable_costs=200)})
+
+# sources
+energysystem.add(
+    Source(
+        label="wind", outputs={bel: Flow(fix=data["wind"], nominal_value=66.3)}
+    )
+)
+
+energysystem.add(
+    Source(label="pv", outputs={bel: Flow(fix=data["pv"], nominal_value=65.3)})
+)
+
+# electricity demand
+energysystem.add(
+    Sink(
+        label="demand_el",
+        inputs={bel: Flow(nominal_value=85, fix=data["demand_el"])},
+    )
+)
+
+# power plants
+energysystem.add(
+    Transformer(
+        label="pp_coal",
+        inputs={bcoal: Flow()},
+        outputs={bel: Flow(nominal_value=20.2, variable_costs=25)},
+        conversion_factors={bel: 0.39},
+    )
+)
+
+energysystem.add(
+    Transformer(
+        label="pp_lig",
+        inputs={blig: Flow()},
+        outputs={bel: Flow(nominal_value=41.8, variable_costs=19)},
+        conversion_factors={bel: 0.41},
+    )
+)
+
+energysystem.add(
+    Transformer(
+        label="pp_gas",
+        inputs={bgas: Flow()},
+        outputs={bel: Flow(nominal_value=41, variable_costs=40)},
+        conversion_factors={bel: 0.50},
+    )
+)
+
+energysystem.add(
+    Transformer(
+        label="pp_oil",
+        inputs={boil: Flow()},
+        outputs={bel: Flow(nominal_value=5, variable_costs=50)},
+        conversion_factors={bel: 0.28},
+    )
+)
+
+# ################################ optimization ###########################
+
+# create optimization model based on energy_system
+optimization_model = Model(energysystem=energysystem)
+
+optimization_model.receive_duals()
+
+# solve problem
+optimization_model.solve(
+    solver=solver, solve_kwargs={"tee": True, "keepfiles": False}
+)
+
+# write back results from optimization object to energysystem
+optimization_model.results()
+
+# ################################ results ################################
+
+# subset of results that includes all flows into and from electrical bus
+# sequences are stored within a pandas.DataFrames and scalars e.g.
+# investment values within a pandas.Series object.
+# in this case the entry data['scalars'] does not exist since no investment
+# variables are used
+data = views.node(optimization_model.results(), "bel")
+data["sequences"].info()
+print("Optimization successful. Showing some results:")
+
+# see: https://pandas.pydata.org/pandas-docs/stable/visualization.html
+node_results_bel = views.node(optimization_model.results(), "bel")
+node_results_flows = node_results_bel["sequences"]
+bel_duals = node_results_flows.pop((("bel", "None"), "duals"))
+node_results_flows = node_results_flows.drop(
+    [(("bel", "demand_el"), "flow"), (("bel", "excess_el"), "flow")], 1
+)
+
+fig, (ax1, ax2) = plt.subplots(2, 1, figsize=(10, 5))
+node_results_flows.plot(ax=ax1, kind="bar", stacked=True, linewidth=0, width=1)
+bel_duals.plot(ax=ax2)
+
```

Comment:
```suggestion
ax1.xaxis = ax2.xaxis
ax1.get_shared_x_axes().join(ax1, ax2)
```

I manage to create a dependency of both x-axis with these two lines. However if I zoom into the plot, it behaves incorrectly. @jnnr could you check this also happens for you?

Reply:
Time axis fixed with commit https://github.com/oemof/oemof-examples/pull/70/commits/0050b7c980e6bcf29fce45c2ce7f94a95c64ac23.